__tests__/main.test.ts

```diff
@@ -0,0 +1,235 @@
+        });
+    });
+});
+
+describe('Run Trivy', () => {
+    let mockFile = {
+        'releaseDownloadedPath': JSON.stringify({ tag_name: 'v1.1.1' })
+    };
+    let cachedTools = {
+        'trivy': true,
+        'dockle': true
+    };
+    mockedFs.__setMockFiles(mockFile);
+    mockedToolCache.__setToolCached(cachedTools);
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    test('Trivy binaries are present in the cache', async () => {
+        const runner = require('../src/trivyHelper');
+        await expect(runner.runTrivy()).resolves.toBe(0);
+        expect(mockedOs.type).not.toHaveBeenCalled();
+        expect(mockedToolCache.find).not.toHaveReturnedWith(undefined);
+        expect(mockedToolCache.downloadTool).toHaveBeenCalledTimes(1);
+    });
+
+    test('Trivy binaries are not present in the cache', async () => {
+        cachedTools['trivy'] = false;
+        mockedToolCache.__setToolCached(cachedTools);
+        const runner = require('../src/trivyHelper');
+
+        await expect(runner.runTrivy()).resolves.toBe(0);
+        expect(mockedOs.type).toHaveBeenCalledTimes(1);
+        expect(mockedToolCache.find).toHaveReturnedWith(undefined);
+        expect(mockedToolCache.downloadTool).toHaveBeenCalledTimes(2);
+    });
+});
+
+describe('Run Dockle', () => {
+    let mockFile = {
+        'releaseDownloadedPath': JSON.stringify({ tag_name: 'v1.1.1' })
+    };
+    let cachedTools = {
+        'trivy': true,
+        'dockle': true
+    };
+    mockedFs.__setMockFiles(mockFile);
+    mockedToolCache.__setToolCached(cachedTools);
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    test('Dockle binaries are present in the cache', async () => {
+        const runner = require('../src/dockleHelper');
+
+        await expect(runner.runDockle()).resolves.toBe(0);
+        expect(mockedOs.type).not.toHaveBeenCalled();
+        expect(mockedToolCache.find).not.toHaveReturnedWith(undefined);
+        expect(mockedToolCache.downloadTool).toHaveBeenCalledTimes(1);
+    });
+
+    test('Dockle binaries are not present in the cache', async () => {
+        cachedTools['dockle'] = false;
+        mockedToolCache.__setToolCached(cachedTools);
+        const runner = require('../src/dockleHelper');
+
+        await expect(runner.runDockle()).resolves.toBe(0);
+        expect(mockedOs.type).toHaveBeenCalledTimes(1);
+        expect(mockedToolCache.find).toHaveReturnedWith(undefined);
+        expect(mockedToolCache.downloadTool).toHaveBeenCalledTimes(2);
+    });
+});
+
+describe('Initialize allowedlist file', () => {
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    test('No allowedlist file is given', () => {
+        const allowedlistHandler = require('../src/allowedlistHandler');
+        expect(allowedlistHandler.init).not.toThrow();
+        expect(mockedFs.existsSync).not.toReturnWith(true);
+        expect(mockedFs.existsSync).toHaveBeenCalledTimes(2);
```

Comment:
Why's it called twice?

Reply:
Once for testing whether `allowedlist.YAML` exists and then to test for `allowedlist.YML`